### PR TITLE
Anthony 1542 issue pinned report

### LINKED ISSUE #1542: Cannot pin a chat if it contains no messages
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/master/CONTRIBUTING.md) for onboarding!
___

**Platform - version:**  
Desktop & Web - 1.0.1-461

**Action Performed (reproducible steps):**
Press the + button to start a new chat with an individual who you have never messaged before. 
Before sending any message, pin the chat.

**Expected Result:**
The chat should be pinned in the chat list on the left.

**Actual Result:**
The pin reflects in the chat list momentarily, but as soon as you click on another chat, it disappears.

https://github.com/Expensify/Expensify/issues/153409
https://www.upwork.com/jobs/~01dc4c143fd4be1b85

### PR DESCRIPTION
**Pull request by anthony-hull**
### Details
Skipping filtering of reports which have no messages but also are pinned

### Fixed Issues
Fixes https://github.com/Expensify/Expensify.cash/issues/1542

### Tests - Unit
`npm run test`
I updated the test file OptionsListUtilsTest.js to count the number of pinned reports. One of the pinned reports doesn't have any messages. 

### Tests - Functional
1. Press the + button to start a new chat with an individual who you have never messaged before.
2. Before sending any message, pin the chat.
3. The chat should be pinned in the chat list on the left.
4. Click on another chat.
5. The chat should still pinned in the chat list on the left.
6. Click back on the chat and unpin the chat.
7. The chat should still be in the list but no longer be pinned.
8. Click on another chat
9. Chat should disappear from the chat list on the left.

### Tested On

- [X] Web
- [ ] Mobile Web
- [ ] Desktop
- [ ] iOS
- [ ] Android

As this is only logic in the data layer I didn't test this on other platforms. I'm relying on the unit tests and

### CODE REVIEW

File: src/libs/OptionsListUtils.js [hunk 1]
```diff
@@ -179,7 +179,9 @@ function getOptions(reports, personalDetails, draftComments, activeReportID, {
         // Skip this entry if it has no comments and is not the active report. We will only show reports from
         // people we have sent or received at least one message with.
         const hasNoComments = report.lastMessageTimestamp === 0;
-        if (!showReportsWithNoComments && hasNoComments && report.reportID !== activeReportID) {
+        const shouldFilterReport = !showReportsWithNoComments && hasNoComments && report.reportID !== activeReportID
+            && !report.isPinned;
```
Comment (deetergp) [inline src/libs/OptionsListUtils.js:183]: This PR looks good overall, just have one minor request: Having this condition by itself on a separate line looks odd. Either put them all on the same line (I think the linter just warns about being over 120 chars, which is fine) or put each condition on its own line.

**Review by deetergp — CHANGES_REQUESTED** [review]

**Review by deetergp — DISMISSED** [review]
Looks good Anthony, thanks!

**Comment by deetergp** [comment]
Looks like you've got a merge conflict to deal with. It looks like you just need to move your test down to the bottom of the page--nothing too complicated.

**Comment by anthony-hull** [comment]
That’s a conflict  my other PR that just merged. It was a race to the finish on both ^^

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Review by deetergp — APPROVED** [review]
Looks good & no more conflicts. Thanks!

**Pull request merged**